Framework/Algorithms/src/MergeRuns/SampleLogsBehaviour.cpp

```diff
@@ -0,0 +1,463 @@
+          "\" defined more than once!");
+    }
+
+    std::shared_ptr<Property> prop;
+    try {
+      prop = std::shared_ptr<Property>(ws->getLog(item)->clone());
+    } catch (std::invalid_argument &) {
+      m_logger.warning() << "Could not merge sample log \"" << item
+                         << "\", does not exist in workspace!" << std::endl;
+      continue;
+    }
+
+    bool isNumeric;
+    double value = 0.0;
+    try {
+      value = ws->getLogAsSingleValue(item);
+      isNumeric = true;
+    } catch (std::invalid_argument &) {
+      isNumeric = false;
+      if (mergeType == time_series) {
+        m_logger.error() << item << " could not be converted to a numeric type"
+                         << std::endl;
+        continue;
+      }
+    }
+
+    if (mergeType == time_series) {
+      try {
+        // See if property exists already - merging an output of MergeRuns
+        prop = std::shared_ptr<Property>(
+            ws->getLog(item + TIME_SERIES_SUFFIX)->clone());
+      } catch (std::invalid_argument &) {
+        // Property does not already exist
+        std::unique_ptr<Kernel::TimeSeriesProperty<double>> timeSeriesProp(
+            new TimeSeriesProperty<double>(item + TIME_SERIES_SUFFIX));
+        std::string startTime = ws->mutableRun().startTime().toISO8601String();
+        timeSeriesProp->addValue(startTime, value);
+        ws->mutableRun().addLogData(
+            std::unique_ptr<Kernel::Property>(std::move(timeSeriesProp)));
+        prop = std::shared_ptr<Property>(
+            ws->getLog(item + TIME_SERIES_SUFFIX)->clone());
+      }
+    } else if (mergeType == list) {
+      try {
+        // See if property exists already - merging an output of MergeRuns
+        prop =
+            std::shared_ptr<Property>(ws->getLog(item + LIST_SUFFIX)->clone());
+      } catch (std::invalid_argument &) {
+        ws->mutableRun().addProperty(item + LIST_SUFFIX, prop->value());
+        prop =
+            std::shared_ptr<Property>(ws->getLog(item + LIST_SUFFIX)->clone());
+      }
+    }
+
+    map[item] = {mergeType, prop, tolerance, isNumeric};
+  }
+}
+
+/**
+ * Creates a vector of tolernaces with the same size as the number of sample
+ * logs for the merge type. If the number of names and tolerances is the same
+ * the vector is filled with the tolerances for each name. If no tolerances were
+ * specified all tolerances are set to -1, and if one tolerance is given all
+ * tolerances are set to that value.
+ *
+ * @param numberNames the number of sample log names
+ * @param tolerances a vector containing strings with the tolerances
+ * @param paramsTolerances a string containing nothing, or tolerances to use
+ * @return a vector of doubles of size numberNames
+ */
+std::vector<double> SampleLogsBehaviour::createTolerancesVector(
+    size_t numberNames, const std::vector<std::string> &tolerances,
+    const std::string paramsTolerances) {
```

Comment:
Yes, that is weird, especially when checking the vector is empty gives the same result! Fixed.